bigquery_etl/cli/query.py

```diff
@@ -1243,6 +1245,73 @@ def validate(
         validate_metadata.validate_datasets(dataset_dir)
 
 
+def _run_initialization_query_for_sample_id(
+    client,
+    query_file,
+    job_config,
+    project_id,
+    dataset,
+    table,
+    addl_templates,
+    sample_id,
+):
+    """Insert data for o given sample_id into the provided table."""
+    print(f"Insert data into table {dataset}.{table} for sample_id={sample_id}...")
+
+    with tempfile.NamedTemporaryFile(mode="w+") as query_stream:
+        query_stream.write(
+            render_template(
+                query_file.name,
+                template_folder=str(query_file.parent),
+                templates_dir="",
+                format=False,
+                **addl_templates,
+            )
+        )
+        query_stream.seek(0)
+        query_content = query_stream.read()
+        client.query(
+            query_content.format(
+                sample_id=sample_id,
+                project_id=project_id,
+                dataset_id=dataset,
+                table_name=table,
+            ),
+            job_config=job_config,
+        ).result()
+
+
+def _run_initialization_query_in_parallel(
+    client,
+    query_file,
+    project_id,
+    dataset_id,
+    table,
+    parallelism,
+    dry_run,
+    addl_templates,
+):
+    job_config = bigquery.QueryJobConfig(dry_run=dry_run, use_query_cache=False)
+
+    with ThreadPool(parallelism) as pool:
+        start = default_timer()
+        # Process all subsets in the query in parallel (eg. all sample_ids).
+        pool.map(
+            partial(
+                _run_initialization_query_for_sample_id,
+                client,
+                query_file,
+                job_config,
+                project_id,
+                dataset_id,
+                table,
+                addl_templates,
+            ),
+            list(range(0, 100)),
+        )
```

Comment:
I believe that we can replace the `_run_initialization_query_for_sample_id` function with `_run_query` here. If we do that, we can use this function for both single-thread and parallelized init. Something like:
```suggestion
        if(parallel_sample_id_run): # Pass this in as a param to `_run_initialization_query_in_parallel` now
          mapped_values = ['-p={"sample_id":"{sample_id}"}' for sample_id in range(0, 100)]
        else:
          mapped_values = ['']
        pool.map(
            partial(
                _run_query,
                query_files=[query_file],
                project_id=project_id,
                public_project_id=None,
                destination_table=None,
                dataset_id=dataset_id,
                addl_templates=addl_templates
            ),
            mapped_values
        )
```

However I would actually prefer to have the parallel init options defined in the `metadata.yaml` rather than in conditional code blocks here...